pyomo/gdp/plugins/bigm.py

```diff
@@ -265,7 +274,7 @@ def _add_transformation_block(self, instance):
             '_pyomo_gdp_bigm_relaxation')
         transBlock = Block()
         instance.add_component(transBlockName, transBlock)
-        transBlock.relaxedDisjuncts = Block(Any)
+        transBlock.relaxedDisjuncts = Block(NonNegativeIntegers)
```

Comment:
Is this the new way to accomplished indexed blocks? Looks like an improvement over `Any`, but in isolation I had to blink a couple times to understand what it's accomplishing.

Reply:
It's not a new way to accomplish indexed blocks: Blocks are still indexable by any Set.  However, since the set-rewrite, all the global Sets (like NonNegativeIntegers) are proper Sets and can be used as indexing sets for components (like Block).  By making the index more restrictive than "Any", @emma58 is making a stronger statement about what a user should expect as a valid index.